src/viztracer/report_builder.py

```diff
@@ -67,18 +68,26 @@ def load_jsons(self) -> None:
                 self.jsons = [get_json(self.data)]
             elif isinstance(self.data, (list, tuple)):
                 self.jsons = []
+                self.invalid_json_paths = []
                 for idx, j in enumerate(self.data):
                     if self.verbose > 0:
                         same_line_print(f"Loading trace data from processes {idx}/{len(self.data)}")
-                    self.jsons.append(get_json(j))
+                    try:
+                        self.jsons.append(get_json(j))
+                    except json.JSONDecodeError:
+                        self.invalid_json_paths.append(j)
+                if len(self.invalid_json_paths) > 0:
+                    self.final_messages.append(("invalid_json", {"paths": self.invalid_json_paths}))
 
     def combine_json(self) -> None:
         if self.verbose > 0:
             same_line_print("Combining trace data")
         if self.combined_json:
             return
-        if not self.jsons:
+        if not self.jsons and not self.invalid_json_paths:
             raise ValueError("Can't get report of nothing")
+        elif not self.jsons:
```

Comment:
Why do you hate nested if so much :) . You are evaluating `self.jsons` twice (performance wise it's not a big deal, but logic wise it's not clean).

Just check `self.jsons` first, and raise different errors based on `self.invalid_json_paths`. Nested if works in many cases.

Reply:
> Why do you hate nested if so much :) . You are evaluating `self.jsons` twice (performance wise it's not a big deal, but logic wise it's not clean).
> 
> Just check `self.jsons` first, and raise different errors based on `self.invalid_json_paths`. Nested if works in many cases.

Sorry, I didn't notice that. And thank you for correcting my bad habit.